suse_migration_services/suse_product.py

```diff
@@ -117,3 +117,33 @@ def get_tag(self, tag):
                 'Parsing XML file {0} failed with: {1}'
                 .format(self.base_product, issue)
             )
+
+    def get_product_name(self):
+        """Get the product name to be migrated to."""
+        migration_product_name = None
+        try:
+            name = self.get_tag('name')[0]
+            arch = self.get_tag('arch')[0]
+            if name and arch:
+                migration_product_name = '/'.join(
+                    [name, self.get_default_target_version(), arch]
+                )
+        except Exception as issue:
+            log.error(
+                'Base product could not be detected: {0}.'.format(issue)
+            )
+
+        return migration_product_name
+
+    def get_default_target_version(self):
+        version_id = None
+        os_release_file = Defaults.get_os_release_file()
+        with open(os_release_file, 'r') as release_file:
+            os_release_data = release_file.read()
+        os_release_data = os_release_data.split(os.linesep)
+        label = 'VERSION_ID'
+        for data in os_release_data:
+            if label in data:
+                version_id = data.split('=')[1]
+                version_id = version_id.replace('"', '')
+        return version_id
```

Comment:
I would implement this more generic. I thought a function that returns class like OSRelease tuple from the defaults space is more generic. For example:

```python
import os
from collections import namedtuple

class Defaults:

    @staticmethod
    def get_os_release():
        with open('/etc/os-release') as handle:
            keys, values = zip(
                *[
                    (key.lower(), value.strip('\'"')) for (key, value) in (
                        line.strip().split('=', 1) for line in
                            handle.read().strip().split(os.linesep)
                    )
                ]
            )
            return namedtuple('OSRelease', keys)(*values)
```

you would then access this by

```python
Defaults.get_os_release().version_id
```

Thoughts ?

Reply:
I thought about a solution like that (returning a dictionary) but changed it as we want _one_ of the fields.
I will implement the approach.

Reply:
I agree, it is usually better to implement what's needed. In this particular case however I can imagine we might need more information from the os release at other parts of the code or for more fine grained selection. Also writing a full parser is less code than looking up a particular element. That's why I thought it's better to be generic from the beginning on. Thanks for changing it